File: C++/majority-element.cpp
class Solution {
public:
    int majorityElement(std::vector<int> &num) {
        int candidate = 0;
        int count = 0;
        for(int i=0; i<num.size(); ++i) {
            if(count==0) {
                candidate = num[i];
                count = 1;
            } else if(num[i]==candidate) {
                ++count;
            } else {
                --count;
            }
        }
        return candidate;
    }
};

File: Python/majority-element.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def majorityElement(self, nums):
        candidate = 0
        count = 0
        for n in nums:
            if count==0:
                candidate = n
                count = 1
            elif candidate==n:
                count += 1
            else:
                count -= 1
        return candidate
